homeassistant/components/whirlpool/strings.json

```diff
@@ -14,5 +14,49 @@
       "unknown": "[%key:common::config_flow::error::unknown%]",
       "no_appliances": "No supported appliances found"
     }
+  },
+  "entity": {
+    "sensor": {
+      "whirlpool_machine": {
+        "state": {
+          "standby": "[%key:common::state::standby%]",
+          "setting": "Setting",
+          "delay_countdown": "Delay Countdown",
+          "delay_paused": "Delay Paused",
+          "smart_delay": "Smart Delay",
+          "smart_grid_pause": "Smart Delay",
+          "pause": "[%key:common::state::paused%]",
+          "running_maincycle": "Running Maincycle",
+          "running_postcycle": "Running Postcycle",
+          "exception": "Exception",
+          "complete": "Complete",
+          "power_failure": "Power Failure",
+          "service_diagnostic_mode": "Service Diagnostic Mode",
+          "factory_diagnostic_mode": "Factory Diagnostic Mode",
+          "life_test": "Life Test",
+          "customer_focus_mode": "Customer Focus Mode",
+          "demo_mode": "Demo Mode",
+          "hard_stop_or_error": "Hard Stop or Error",
+          "system_initialize": "System Initialize",
+          "cycle_filling": "Cycle Filling",
+          "cycle_rinsing": "Cycle Rinsing",
+          "cycle_sensing": "Cycle Sensing",
+          "cycle_soaking": "Cycle Soaking",
+          "cycle_spinning": "Cycle Spinning",
+          "cycle_washing": "Cycle Washing",
+          "door_open": "Door Open"
+        }
+      },
+      "whirlpool_tank": {
+        "state": {
+          "unknown": "Unknown",
+          "empty": "Empty",
+          "25%": "25%",
+          "50%": "50%",
+          "100%": "100%",
```

Comment:
While finishing up adding checks to our translations in #114877, I came across this PR / implementation, which uses invalid translation keys.

```
Integration whirlpool:
Error: R] [TRANSLATIONS] Invalid strings.json: Invalid translation key '25%', need to be [a-z0-9-_]+ and cannot start or end with a hyphen or underscore. for dictionary value @ data['entity']['sensor']['whirlpool_tank']['state']. Got {'unknown': 'Unknown', 'empty': 'Empty', '25%': '25%', '50%': '50%', '100%': '100%', 'active': '[%key:common::state::active%]'}
```

The `%` cannot be part of a translation key, it has meaning in our translation platform as well.

@mkmer Could you take a look at this?

Reply:
So much to learn...  PR https://github.com/home-assistant/core/pull/85849